metabase/metabase
# Enable deleting groups and members with authentication mapping group deletion

Partially resolves https://github.com/metabase/metabase/issues/27212

### How to Test

📺 **Loom Demo**: https://www.loom.com/share/d90a32af70114d07bbf37b946e3b009b

1. In http://localhost:3000/admin/people/groups, create a few groups, and populate them with users
2. In http://localhost:3000/admin/settings/authentication/ldap, scroll all the way down and click on `Edit Mappings`
3. Create mappings, and populate them with zero to several groups each.
4. Save
5. Remove mappings
6. If a mapping is not linked to any groups, the mapping will simply disappear from the list. 
7. If a mapping is linked to one group, a modal will be displayed, with options on what to do about the groups in the mappings and copy using "group" in the singular from whenever appropriate.
8. If a mapping is linked to more than one group, a modal will be displayed, with options on what to do about the groups in the mappings and copy using "group" in the plural from whenever appropriate.

9. Save

Back in http://localhost:3000/admin/people/groups, you should:
1. no longer see the groups you had marked for deletion (Admin should still be there)
2. no longer see users in groups you had marked for emptying of users (Admin should still have the same users as before)

#### Corner cases 

1. If you create a mapping and click to Remove it before saving, no modal will appear. The groups mapped will not be affected.

### What will follow in upcoming PRs

1. The Mapping interaction will be inlined, removed from the Modal.
2. Unit tests for GroupMappingsWidget will be added.
3. Break-down GroupMappingsWidget.
